traits/observers/expression.py

```diff
@@ -73,10 +58,8 @@ def __or__(self, expression):
         new_expression : traits.observers.expression.Expression
         """
         if self == expression:
-            return self._copy()
-        new = Expression()
-        new._prior_expression = _ParallelExpression([self, expression])
-        return new
+            return self
```

Comment:
Could we drop this branch? Is it required for correctness, or is it more for optimization?

Reply:
(And if it's required for correctness, what happens with something like `expr1 | expr2 | expr1`?)

Reply:
This is an optimization and yes we can drop it.